### ISSUE TITLE
AvaloniaXPlatExample.Web - Bar Chart does not display

### ISSUE THREAD

**Issue by houstonhaynes** [beto-rodriguez/LiveCharts2#1397]
**Describe the bug**
Using 2.0.0-rc2 in .NET8 the bar chart in the ReactiveElmish.Avalonia sample app does not render in web (wasm) 

**To Reproduce**
1. Go to [ReactiveElmish Repo](https://github.com/JordanMarr/ReactiveElmish.Avalonia)
2. clone
3. nav to XPlat Sample folder
4. restore workload in .Web
5. build/debug

**Expected behavior**
When navigating to the "Chart" view using the button in the top nav the bar chart should initialize and animate/display a range of randomized values.

**Screenshots**
![image](https://github.com/beto-rodriguez/LiveCharts2/assets/8174976/b4d20be1-1822-4846-9d40-c9a8ff01428c)

**Desktop (please complete the following information):**
 - OS: WinOS
 - Browser Chrome
 - Version 2.2.0-rc2

**Additional context**
.Desktop on the same build works as expected. 

![image](https://github.com/beto-rodriguez/LiveCharts2/assets/8174976/31d709b7-0008-4f93-bcfe-58cb74225a02)

**Comment by daz10000** [comment]
I did the update as well last night and still no dice - it doesn't say anything either on the console.  I was going to instrument the code.  One thought - they seem to offer a few versions of Livechart that are platform specific (libraries in nuget) - Avalonia is one, blazor another etc.  They do say it works on wasm, but wondering if you need to add a specific library to get it working there.

**Comment by houstonhaynes** [comment]
Thanks I was wondering whether "going back to go forward" was worth considering.  

It definitely worked when I logged the Elmish.Avalonia issue on lag in response - which led Jordan down the path of adding the terminate to dispose of subscriptions on exiting a view. I'd love to be able to find the combination of tooling that will get this "off the ground" but I'm still in learning mode traversing this stack.

**Comment by daz10000** [comment]
This might be a good question for @beto-rodriguez - Hi!   We have a nice demo Avalonia cross platform app using your lovely Livecharts, and the demo seems to work for Desktop, Android so far but there's just a blank space when we are running Avalonia via wasm/web.  Is there anything special we need to do for WASM?  I saw you have Blazor support and also Avalonia support - we are running avalonia compiled down to wasm and as you can see above, there's just a blank space where the graph would appear (remainder of app working).  Any tips for debugging appreciated

**Comment by beto-rodriguez** [comment]
Interesting... I need to check this, maybe there is a breaking change in Avalonia, it used to work, I am sure of that... it is just strange that there is no error in the console.

**Comment by daz10000** [comment]
To be clear it is working on some avalonia platforms - desktop and I believe android. The wasm/web version doesn’t display and console messages aren’t always captured - I have had success seeing them on some browsers in dev mode when the code is writing explicitly to console but I suspect some processes die in background on exception Am 12/14/23 um 11:01 PM schrieb Alberto Rodríguez ***@***.***>:﻿
Interesting... I need to check this, maybe there is a breaking change in Avalonia, it used to work, I am sure of that... it is just strange that there is no error in the console.

**Comment by daz10000** [comment]
I was thinking the best way to troubleshoot might be to build the chart tools locally, link it directly into the project and just litter it with print statements / try /except blocks till I find out which part isn't work - it's something WASM related I would bet, as the charts seems to work fine on other platforms inside Avalonia.  Do you have them running elsewhere on a browser with wasm?

**Comment by stevemonaco** [comment]
.NET 8 made some breaking changes to WASM. I was able to correctly run this app under .NET 7 (release and debug builds, both under the older LiveCharts2 version and 2.0.0-rc2).

However, when I updated to .NET 8, then I could replicate the invisible bar chart (both debug/release builds). Downgrading LiveCharts2 did not help. Changing TargetFramework to `net8.0-browser` didn't help. I think trimming defaults changed in .NET 8 WASM, so I would check there. I wasn't able to trigger any trim warnings, but I'm not very familiar with F# or WASM.

**Comment by stevemonaco** [comment]
I also noticed that you aren't using compiled bindings. I would suggest enabling them by default (`<AvaloniaUseCompiledBindingsByDefault>true</AvaloniaUseCompiledBindingsByDefault>` in the fspoj). You can't really use reflection bindings safely here, AFAIK, because of the trimming and ultimately you probably want NativeAOT. I tried enabling them just for the ChartView, but it didn't seem to yield results.

**Comment by houstonhaynes** [comment]
Just confirming here that we didn't see a difference with WASM regardless of compiled bindings settings. One of the major pivots in the ReactiveElmish.Avalonia library was shifting to compiled bindings *on* by default for all the reasons. 👍

**Comment by daz10000** [comment]
I've seen problems with trimming previously for AOT compilation for non wasm projects - it throws out some dependency that's not obvious.  That said, I can't see where that's configured for this template family.  I haven't tested net7 yet either, but that seems like a reasonable hypothesis that it's a runtime change since it was working happily earlier.

**Comment by houstonhaynes** [comment]
Yeah I had trouble running .NET7 with F#8 so I kinda gave up - could have been me not minding some touchpoint to make those two work. So I didn't drill too hard into .NET7 as the F#8 binding method is the new hotness and I didn't feel like re-writing it in the previous style. 😞  There's no AOT here that I know of - I planned on looking at it for some OpenFaaS-driven serverless stuff I have running on a utility server, but more out of curiosity than anything else. The only other place I'm looking at for AOT is WildernessLabs microcontrollers. 🤖

**Comment by MajesticBevans** [comment]
Just wanted to log that I am making a C# Avalonia XPLAT project with LiveCharts and while the desktop build works fine, I'm also getting a similar issue that the chart does not display in the wasm browser build.

**Comment by beto-rodriguez** [comment]
For me currently it works, I added a WASM sample for Avalonia in this repo and it works, not sure if the fix was on LiveCharts or in the Avalonia side, but it works, I'll keep this open for a while just to ensure the issue is no longer present, please let me know if the issue is also gone for you.

**Comment by houstonhaynes** [comment]
I'm just now turning back to a desktop/systray project that has Avalonia in it and I'll check it out. Thanks!

**Comment by beto-rodriguez** [comment]
Closing as stale.

A working Avalonia WASM sample lives in this repo at samples/AvaloniaSample/Platforms/AvaloniaSample.Browser on net10.0-browser, and the chart renders there. The original symptom (blank canvas, no console error, WASM-only, surfaced when moving to .NET 8) was a hosting/trimming/native-asset issue on the Avalonia + SkiaSharp WASM side, not something LiveCharts can fix from its side — the core and SkiaSharp view have no WASM-specific code paths.

If anyone is still hitting this on a current LiveCharts + Avalonia version, please open a fresh issue with a minimal repro against the latest packages and we'll take another look.
